### PR TITLE
Training with largest batch size

### PR DESCRIPTION
**Pull request by loganhart420**
This is a new fit option to train on the largest batch size your gpu can fit. It starts with the largest batch size(defaults to 2048) and cuts the batch size in half until no memory error occurs, then it runs training as normal

### CODE REVIEW

**Comment by loganhart420** [comment]
I still need to push openai finder. Other stuff taking priority so won' t be pushed till Saturday. Just need to write unitest for code.

File: trainer/trainer.py [hunk 1]
```diff
@@ -1483,10 +1485,33 @@ def _fit(self) -> None:
                 self.save_best_model()
             self.callbacks.on_epoch_end(self)
 
-    def fit(self) -> None:
+    def _fit_with_largest_batch_size(self, starting_batch_size=2048):
```
Comment (erogol) [inline trainer/trainer.py:1488]: How about making this an explicit call with a separate function like Trainer.fit_with_batch_finder()

**Review by erogol — CHANGES_REQUESTED** [review]

File: trainer/trainer.py (hunk 1 above)
Comment (loganhart420) [reply trainer/trainer.py:1488]: I like that idea actually. smart

**Comment by loganhart420** [comment]
I'm still messing with the openai paper finder but we can merge this one with main so I'm going to make a separate PR once I'm done. I'm honestly trying to see if how many times the openai finder doesn't just pick the highest batch size for training. It has on some tasks so if stays like that we don't need it

**Comment by erogol** [comment]
There are duplicate commits. You need to rebase your code before further changes.

**Comment by loganhart420** [comment]
> @loganhart420 Can you add a short "how to" to the README and some info on what this is in this issue?

what like an example?

**Pull request closed without merging**